File: java/parser/src/main/java/gov/lbl/parser/parser/Array_subscriptsVisitor.java
package gov.lbl.parser.parser;

import java.util.List;

import gov.lbl.antlr4.visitor.modelicaBaseVisitor;
import gov.lbl.antlr4.visitor.modelicaParser;
import gov.lbl.parser.domain.Array_subscripts;
import gov.lbl.parser.domain.Subscript;

import static java.util.stream.Collectors.toList;


public class Array_subscriptsVisitor extends modelicaBaseVisitor<Array_subscripts> {
    @Override
    public Array_subscripts visitArray_subscripts(modelicaParser.Array_subscriptsContext ctx) {
        SubscriptVisitor subscriptVisitor = new SubscriptVisitor();
        List<Subscript> subscripts = ctx.subscript()
                .stream()
                .map(subscript -> subscript.accept(subscriptVisitor))
                .collect(toList());
        
        return new Array_subscripts(subscripts);
    }
}

File: jsParser/parser/Array_subscriptsVisitor.js
const modelicaVisitor = require('../antlrFiles/modelicaVisitor').modelicaVisitor
const Array_subscripts = require('../domain/Array_subscripts')

const SubscriptVisitor = require('./SubscriptVisitor')

class Array_subscriptsVisitor {
  constructor () {
    modelicaVisitor.call(this)
    return this
  }

  visitArray_subscripts (ctx) {
    const subscripts = []

    if (ctx.subscript()) {
      const subscriptVisitor = new SubscriptVisitor.SubscriptVisitor()
      ctx.subscript().forEach(sub => {
        subscripts.push(subscriptVisitor.visitSubscript(sub))
      })
    }
    return new Array_subscripts.Array_subscripts(subscripts)
  }
};

Array_subscriptsVisitor.prototype = Object.create(modelicaVisitor.prototype)

exports.visitArray_subscripts = this.visitArray_subscripts
exports.Array_subscriptsVisitor = Array_subscriptsVisitor
